### ISSUE TITLE
Code Completion for the CLI would be a big advantage

### ISSUE THREAD

**Issue by james-nesbitt** [wunderkraut/radi-cli#5]
Most tools provide at least bash completion these days. Some ZSH completion would be a high priority for us in wunder.

**Comment by james-nesbitt** [comment]
The CLI uses a common golang cli helper base : https://github.com/urfave/cli

This helper already provides some functionality for bash completion (https://github.com/urfave/cli#bash-completion) but we have not implemented that.

That said, there is a limit to what completion we should offer, and that limit is not clear yet.

For example, good completions would be:

1. autocomplete commands and parameters

Wheras risky completions would be:

1. autocomplete upcloud server name : too easily stalled with a timeout.

**Comment by james-nesbitt** [comment]
Unfortunately ZSH completion won't be as easy for us, but we can try to get it integrated at some point.

**Comment by james-nesbitt** [comment]
it appears that the newer versions of our cli have an abstraction for code completion.  We can now assign completion functions to each command, which means that we can probably more easily write zsh-completion.

**Comment by james-nesbitt** [comment]
Our upsrtream CLI library definitely supports zsh and bash completion.  What is the priority on this task?

https://github.com/urfave/cli/tree/v2#shell-completion

**Comment by tharna** [comment]
This is nice to have, but only after we have stable local development environments up and running.

**Comment by iler** [comment]
I agree with Janne here. For longer run this is important but for now let's concentrate on getting the MVP out :)
